src/index.ts

```diff
@@ -453,13 +453,17 @@ export type Token = string | Key;
 function regexpToRegexp(path: RegExp, keys?: Key[]): RegExp {
   if (!keys) return path;
 
-  // Use a negative lookahead to match only capturing groups.
-  const groups = path.source.match(/\((?!\?)/g);
+  const groups = path.source.match(/\((\?<.*?>)?(?!\?)/g);
```

Comment:
```suggestion
  const groups = path.source.match(/\((?:\?<(.*?)>)?(?!\?)/g);
```

Something like this will resolve the comment you have below.

Reply:
Yeah, non-capturing groups, I tried that. Had no luck so I'm guessing, I made some mistake :) 

Will try this :)

Reply:
> Something like this will resolve the comment you have below.

That didn't work 
```
$ curl http://localhost:1337/testRegex/testData/extraStuff -w "\n"
{"0":"testRegex","2":"extraStuff","(?<groupname>":"testData"}
```

Reply:
Oh yeah, that makes sense. It's happening with `.match` which is only getting the matches and not the groups. We probably need to use `regexp.exec` in a loop instead.

There's also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll but it's not supported on earlier node.js versions or browsers.

Reply:
Using `exec` worked :)

Looping through `exec` results doesn't look nice tho.
Any idea how to make it nicer?